### ISSUE TITLE
CLDF 1.3 and alternative translations

### ISSUE THREAD

**Issue by johenglisch** [dictionaria/pydictionaria#28]
CLDF 1.3 now has official support for multiple translations in examples (https://github.com/cldf/cldf/issues/154), so maybe it's time to get rid of that janky *alt_translation1*/*alt_translation2* business.

This would change the database schema (and the schema for the *md.json*), so we're looking at a major release for pydictionaria and all dictionaries with alt. translations (14 if I counted correctly).

 * [ ] Generate alternate translations from *pydictionaria* based on the standard.
 * [ ] Make Dictionaria webapp able to read the new format.
 * [ ] Bump up *pycldf* requirement to [v1.37](https://github.com/cldf/pycldf/releases/tag/v1.37.0).
 * [ ] Actually write a proper README (documenting the changes and the package in general).
 * [ ] Major release of pydictionaria.
 * [ ] Make all dictionaries depend on pydictionaria 3.0.
 * [ ] Adapt their *md.json*.
 * [ ] Regen.
 * [ ] Major release for all the things.
 * [ ] Redeploy web app.

**Comment by johenglisch** [comment]
Okay, when I woke up this morning I noticed I completely missed the fact that alternative translations affect senses as well (hurray to sleep).  AFAICT CLDF 1.3 doesn't account for this explicitly (unless I missed something).

So, I guess we could handle senses analogous to how CLDF 1.3 treats corpora, i.e. a table like this:

    | ID | Entry_ID | Description | Scientific_Name | alt_translation1 | alt_translation2 |
    |----+----------+-------------+-----------------+------------------+------------------|
    |  1 |        1 | lime tree   | tilia           | Linde            | липа             |
    |  2 |        2 | sun         |                 | Sonne            | солнце           |
    |  3 |        2 | day         |                 | Tag              | день             |

would get reshaped into this:

    | ID | Entry_ID | Description | Scientific_Name | Meta_Language_ID | Full_Sense |
    |----+----------+-------------+-----------------+------------------+------------|
    |  1 |        1 | lime tree   | tilia           | eng              |            |
    |  2 |        1 | Linde       |                 | deu              |          1 |
    |  3 |        1 | липа        |                 | rus              |          1 |
    |  4 |        2 | sun         |                 | eng              |            |
    |  5 |        2 | Sonne       |                 | deu              |          4 |
    |  6 |        2 | солнце      |                 | rus              |          4 |
    |  7 |        2 | day         |                 | eng              |            |
    |  8 |        2 | Tag         |                 | deu              |          7 |
    |  9 |        2 | день        |                 | rus              |          7 |

Changing the Sense Table is potentially more distruptive downstream than the Example Table, so I'd like to hear people's thoughts.

@xrotwang Does this look reasonable?

@fredericblum @lingulist How hard would this break your code?  I suspect not very -- all you'd have to do is filter out the senses you don't want, right?  E.g.:

    only_english = [s for s in senses if s.get('Meta_Language_ID') == 'eng']
    only_primary = [s for s in senses if not s.get('Full_Sense')]

**Comment by xrotwang** [comment]
Yes, I think this does look reasonable. Arguably, for senses this kind of data (i.e. alternative translations) should be somehow offloaded to something like Concepticon. But realistically, this is not (fully) going to happen.

This case not being explicitly included in CLDF 1.3 is probably another effect of the `Dictionary` module being one of the modules without a "patron" :)
